src/PdfParser/Type/PdfStream.php

```diff
@@ -31,11 +31,11 @@ class PdfStream extends PdfType
      *
      * @param PdfDictionary $dictionary
      * @param StreamReader $reader
-     * @param PdfParser $parser Optional to keep backwards compatibility
+     * @param PdfParser|null $parser Optional to keep backwards compatibility
      * @return self
      * @throws PdfTypeException
      */
-    public static function parse(PdfDictionary $dictionary, StreamReader $reader, PdfParser $parser = null)
+    public static function parse(PdfDictionary $dictionary, StreamReader $reader, $parser = null)
```

Comment:
Can you please ensure that `$v->parser` (in line 43) is only set if `$parser` is an instance of `PdfParser`?

Reply:
done